src/integer/z/distance.rs

```diff
@@ -0,0 +1,120 @@
+// Copyright © 2023 Niklas Siemer
+//
+// This file is part of qFALL-math.
+//
+// qFALL-math is free software: you can redistribute it and/or modify it under
+// the terms of the Mozilla Public License Version 2.0 as published by the
+// Mozilla Foundation. See <https://mozilla.org/en-US/MPL/2.0/>.
+
+//! This module contains the implementation of the [`Distance`] trait for [`Z`].
+
+use super::Z;
+use crate::{
+    integer_mod_q::{Modulus, Zq},
+    macros::for_others::{implement_for_others, implement_for_owned},
+    traits::Distance,
+};
+
+impl Distance<&Z> for Z {
+    type Output = Z;
+
+    /// Computes the absolute distance between two [`Z`] instances.
+    ///
+    /// Parameters:
+    /// - `other`: specifies one of the [`Z`] values whose distance
+    /// is calculated to `self`
+    ///
+    /// Returns the absolute difference, i.e. distance between the two given [`Z`]
+    /// instances as a new [`Z`] instance.
+    ///
+    /// # Example
+    /// ```
+    /// use qfall_math::integer::Z;
+    /// use qfall_math::traits::*;
+    ///
+    /// let a = Z::from(-1);
+    /// let b = Z::from(10);
+    ///
+    /// let distance_0 = a.distance(5);
+    /// let distance_1 = a.distance(10);
+    ///
+    /// # assert_eq!(Z::from(6), distance_0);
+    /// # assert_eq!(Z::from(11), distance_1);
+    /// ```
+    fn distance(&self, other: &Z) -> Self::Output {
+        let difference = other - self;
+        difference.abs()
+    }
+}
+
+implement_for_owned!(Z, Z, Distance);
+implement_for_others!(Z, Z, Distance for u8 u16 u32 u64 i8 i16 i32 i64 Modulus Zq);
```

Comment:
Do you want Zq to be implemented by this macro? I personally would not be quite shur how the distance is measured in theis case, hence I think that the simple Macro implementation seems off here. I would remove it, or implement it separately with a dedicated description on the expected output
```suggestion
implement_for_others!(Z, Z, Distance for u8 u16 u32 u64 i8 i16 i32 i64 Modulus);
```